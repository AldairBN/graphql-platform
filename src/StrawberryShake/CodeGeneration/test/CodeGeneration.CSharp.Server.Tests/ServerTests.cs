using System.IO;
using System.Threading;
using System.Threading.Tasks;
using Snapshooter.Xunit;
using Xunit;

namespace StrawberryShake.CodeGeneration.CSharp;

public class ServerTests
{
    [Fact]
    public async Task Generate_StarWarsAsync()
    {
        // arrange
        using var cts = new CancellationTokenSource(3000000);
        var readStream = new TestStream();
        var writeStream = new TestStream();
        await using var client = new CSharpGeneratorClient(readStream, writeStream);
        await using var server = new CSharpGeneratorServer(writeStream, readStream);
        var configFile = FilePath(".graphqlrc.json");
        var documents = new[]
        {
            FilePath("ChatGetPeople.graphql"),
            FilePath("Schema.extensions.graphql"),
            FilePath("Schema.graphql")
        };

        // act
        GeneratorRequest request = new(configFile, documents);
        GeneratorResponse response = await client.GenerateAsync(request, cts.Token);

        // assert
        response.MatchSnapshot();
    }

    private static string FilePath(string name)
        => Path.Combine("__resources__", name);
}
