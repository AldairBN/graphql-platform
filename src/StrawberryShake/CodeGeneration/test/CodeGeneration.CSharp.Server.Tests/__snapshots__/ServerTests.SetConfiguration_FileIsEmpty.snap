{
  "Errors": [
    {
      "Message": "The `fileName` cannot be null or empty."
    }
  ]
}
